webgpu.yml

```diff
@@ -226,21 +226,34 @@ enums:
         doc: |
           TODO
   - name: callback_mode
-    doc: |
-      TODO
+    doc: The callback mode controls how a callback for an asynchronous operation may be fired. See @ref Asynchronous-Operations for how these are used.
     entries:
       - name: wait_any_only
         value: 0x0001
         doc: |
-          TODO
+          Callbacks created with `WGPUCallbackMode_WaitAnyOnly`:
+          - fire when the asynchronous operation's future is passed to a call to `::wgpuInstanceWaitAny`
+            AND the operation has already completed or it completes inside the call to `::wgpuInstanceWaitAny`.
       - name: allow_process_events
         value: 0x0002
         doc: |
-          TODO
+          Callbacks created with `WGPUCallbackMode_AllowProcessEvents`:
+          - fire when the asynchronous operation's future is passed to a call to `::wgpuInstanceWaitAny`
+            AND the operation has already completed or it completes inside the call to `::wgpuInstanceWaitAny`.
+          - fire inside a call to `::wgpuInstanceProcessEvents` if the asynchronous asychronous operation is complete.
       - name: allow_spontaneous
         value: 0x0003
         doc: |
-          TODO
+          Callbacks created with `WGPUCallbackMode_AllowSpontaneous`:
+          - fire when the asynchronous operation's future is passed to a call to `::wgpuInstanceWaitAny`
+            AND the operation has already completed or it completes inside the call to `::wgpuInstanceWaitAny`.
+          - fire inside a call to `::wgpuInstanceProcessEvents` if the asynchronous asychronous operation is complete.
+          - **may** fire spontaneously on an arbitrary or application thread, when the WebGPU implementations discovers that the asynchronous operation is complete.
+
+            Implementations _should_ fire spontaneous callbacks as soon as possible.
+
+          Because spontaneous callbacks may fire at an arbitrary time on an arbitrary thread, applications should take extra care when acquiring locks or mutating state inside the callback. It undefined behavior to re-entrantly call into the webgpu.h API if the callback fires while inside the callstack of another webgpu.h function.
```

Comment:
I thought we wouldn't need this reentrancy rule, though it is tricky to implement, so I'm OK with keeping it for now.

Reply:
I think we would like to make it be the case, but not sure how doable it is yet. So I figured we have a note about this for now.